php/libraries/Candidate.class.inc

```diff
@@ -621,11 +621,22 @@ class Candidate
         $db      = $factory->database();
 
         $candID = $this->getCandID();
-
+        // Visits may occur on the same date, so ordering by Date_Visit
+        // or selecting by MIN(Date_Visit) will not produce a unique result
+        // as required by pselectOne
+        // This selects the smallest session ID, in the event that there
+        // are multiple visits starting on the same day
         $query  = "SELECT Visit_label
-            FROM session s JOIN candidate c ON (c.CandID = s.CandID)
-            WHERE c.CandID=:cid AND Date_Visit IS NOT NULL
-            ORDER BY Date_Visit";
+             FROM session 
+             WHERE ID =  
+              (SELECT MIN(ID) FROM session
+                WHERE Date_Visit =
+                  (SELECT MIN(Date_Visit) FROM session 
+                    WHERE CandID=:cid 
+                    AND Date_Visit IS NOT NULL
+                  ) 
+                AND CandID=:cid
+             )";
```

Comment:
I think it would be better to just not use the pselectOne helper and grab everything for the CandID, then find the earliest date in PHP than to have a subselect within a subselect within a query.

Reply:
faster to execute directly in MySQL than in PHP. Less data to transmit between the 2 and less memory used in Apache

Reply:
The LIMIT clause can help for that.
What about : `SELECT Visit_label FROM session WHERE CandID = 595267 AND Date_visit IS NOT NULL ORDER BY Date_visit, ID ASC LIMIT 1`